build.gradle.kts

```diff
@@ -0,0 +1,169 @@
+
+    // Plugin dependencies
+    compileOnly("com.sk89q.worldedit:worldedit-core:7.2.19")
+    compileOnly("com.sk89q.worldedit:worldedit-bukkit:7.2.19")
+    compileOnly("com.gmail.nossr50.mcMMO:mcMMO:2.1.230")
+    compileOnly("me.clip:placeholderapi:2.11.5")
+    compileOnly("me.minebuilders:clearlag-core:3.1.6")
+    compileOnly("com.github.LoneDev6:itemsadder-api:3.6.1")
+    compileOnly("net.imprex:orebfuscator-api:5.4.0")
+
+
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+    testImplementation("org.junit.jupiter:junit-jupiter:5.10.2")
+    testImplementation("org.mockito:mockito-core:5.10.0")
+    testImplementation("org.slf4j:slf4j-simple:2.0.9")
+    testImplementation("com.github.seeseemelk:MockBukkit-v1.20:3.65.0")
+
+    // Override MockBukkit's Paper to a pinned slightly older version
+    // This is because MockBukkit currently fails after this PR: https://github.com/PaperMC/Paper/pull/9629
+    testImplementation("io.papermc.paper:paper-api:1.20.4-R0.1-20240205.114523-90")
+}
+
+group = "com.github.slimefun"
+version = property("project.version").toString()
+description = "Slimefun"
+
+java.toolchain.languageVersion.set(JavaLanguageVersion.of(targetJavaVersion))
+
+java {
+    withSourcesJar()
+}
+
+tasks {
+    withType<JavaCompile>().configureEach {
+        options.release = targetJavaVersion
+        options.encoding = encoding
+
+        // package info files are only important for Javadocs
+        // We can safely exclude them from the final jar -->
+        exclude("**/package-info.java")
+    }
+    withType<Javadoc>().configureEach {
+        options.encoding = encoding
+    }
+    javadoc {
+        setDestinationDir(file("javadocs"))
+        options {
+            title = "Slimefun4 - Javadocs"
+            windowTitle = "Slimefun4 - Javadocs"
+            // doclet = "org.gradle.external.javadoc.StandardJavadocDocletOptions"
+            // FIXME offline links, groups, and additional options.
+        }
+
+    }
+    assemble {
+        dependsOn(shadowJar)
+
+    }
+    jar {
+        archiveClassifier.set("original")
+        archiveVersion.set("v${project.version}")
+    }
+    shadowJar {
+        archiveClassifier.set("")
+        archiveVersion.set("v${project.version}")
+
+        // Relocations
+        val destination = "io.github.thebusybiscuit.slimefun4.libraries"
+        relocate("io.github.bakedlibs.dough", "$destination.dough")
+        relocate("io.papermc.lib", "$destination.paperlib")
+        relocate("org.apache.commons.lang", "$destination.commons.lang")
+        // Merge the META-INF files
+        mergeServiceFiles()
+    }
+    processResources {
+
+        include("plugin.yml")
+        include("config.yml")
+        include("item-models.yml")
+        include("wiki.json")
+        include("languages/translators.json")
+        include("tags/*.json")
+        include("biome-maps/*.json")
+        include("languages/**/*.yml")
+        filesMatching("plugin.yml") {
```

Comment:
This can be replaced with the `plugin-yml` gradle plugin, tho idk how much others will want it

Reply:
Yes, haven't gotten to that part yet so I left it as we used to with maven resource filtering.